### PR TITLE
refactor: Improve auth sessions using context

### PR DESCRIPTION
**Pull request by Exzotic5485**
Improves the current auth / session code by using react context and a "useSession()" hook that can be used anwhere (client components) in the app.

### CODE REVIEW

**Review by tkbstudios — CHANGES_REQUESTED** [review]
Seems fine by me, I just requested one change, maybe? (I'm not sure)

**Comment by tkbstudios** [comment]
@okunamayanad review?

**Comment by tkbstudios** [comment]
Noice stuff man

File: src/app/dashboard/page.tsx [hunk 1]
```diff
@@ -268,100 +261,80 @@ export default function Dashboard({
...
-        fetchTransactions();
-      } else {
-        setTimeout(() => {
-          // wait for sonner to load
-          toast.error("You are not logged in. Redirecting to home page.");
-          router.push("/");
-        }, 100);
-      }
-    })();
   }, [router]);
```
[the remark this answers is not in the document]
Comment (okunamayanad) [reply src/app/dashboard/page.tsx]: No it won't. The issue is not releated to router, it's releated to `fetchTransactions` not being in the dependency array. But I see changes in that too and it is actually in fact fixed!

**Pull request merged**